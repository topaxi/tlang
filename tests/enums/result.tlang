let ok = Result::Ok(10);
let err = Result::Err(5);

ok.is_ok() |> log();
ok.is_err() |> log();

err.is_ok() |> log();
err.is_err() |> log();

ok.unwrap() |> log();
err.unwrap() |> log(); // This should panic
