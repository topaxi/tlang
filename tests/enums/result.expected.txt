true
false
false
true
10


thread 'main' panicked at crates/tlang_runtime/tlang_memory/src/state.rs:276:9:
Called unwrap on Err
  at root 10:1

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
